### PR TITLE
upd zlib to 1.3

### PR DESCRIPTION
**Pull request by ekomarova**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This PR is designed to update zlib to 1.3 and address https://nvd.nist.gov/vuln/detail/CVE-2023-45853

### CODE REVIEW

**Comment by ekomarova** [comment]
@conda-forge-admin, please rerender

**Comment by ekomarova** [comment]
@groutr @msarahan @isuruf @ocefpaf Please review

**Comment by ocefpaf** [comment]
LGTM, the appveyor hook is there due to an error. B/c zlib is fundamental do many packages I'd love for @isuruf's input before we merge this one.

**Comment by Neustradamus** [comment]
@ekomarova: Do not forget the CVE patch too before a 1.3.1 build.

Linked to:
- https://github.com/madler/zlib/issues/868

**Comment by ekomarova** [comment]
> @ekomarova: Do not forget the CVE patch too before a 1.3.1 build.
> 
> Linked to:
> 
> * [Please release 1.3.1 to fix CVE-2023-45853 (9.8 CRITICAL) madler/zlib#868](https://github.com/madler/zlib/issues/868)

Oh, it looks like I missed that this will only be fixed in 1.3.1

**Comment by ekomarova** [comment]
@conda-forge-admin, please rerender

**Comment by ekomarova** [comment]
@groutr @msarahan @isuruf @ocefpaf Please review

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,7 +9,9 @@ source:
   url:
     - http://zlib.net/zlib-{{ version }}.tar.gz
     - https://gnupg.org/ftp/gcrypt/zlib/zlib-{{ version }}.tar.gz
-  sha256: b3a24de97a8fdbc835b9833169501030b8977031bcb54b3b3ac13740f846ab30
+  sha256: ff0ba4c292013dbc27530b3a81e1f9a813cd39de01ca5e0f8bf355702efa593e
+  patches:
+    - CVE-2023-45853.patch
```
Comment (ocefpaf) [inline recipe/meta.yaml:15]: Can you add a comment with the PR (https://github.com/madler/zlib/pull/843) and some text reminding us to drop it in the new release?

**Pull request merged**